src/App.tsx

```diff
@@ -1,61 +1,125 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getBiggestId = (todo: Todo[]): number => {
+  return Math.max(...todo.map(newTodo => newTodo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setNewUserId] = useState(0);
+  const [todoList, setTodoList] = useState(todos);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const createNewTodo = (title: string, userId: number) => {
+    const newTodo: Todo = {
+      id: getBiggestId(todos),
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    setTodoList([...todoList, newTodo]);
+
+    setNewTitle('');
+    setNewUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setShowTitleError(!newTitle);
+    setShowUserError(!newUserId);
+
+    if (newTitle && newUserId) {
+      createNewTodo(newTitle, newUserId);
```

Comment:
Consider renaming `createNewTodo` to `handleCreateNewTodo` to better describe what the function does. 🤔